cms/models/pluginmodel.py

```diff
@@ -334,21 +334,26 @@ def copy_plugin(self, target_placeholder, target_language, parent_cache, no_sign
     @classmethod
     def fix_tree(cls, destructive=False):
         """
-        Fixes the plugin tree by first calling treebeard fix_tree and the recalculating
-        the correct position property for each plugin.
+        Fixes the plugin tree by first calling treebeard fix_tree and the
+        recalculating the correct position property for each plugin.
         """
         from cms.utils.plugins import reorder_plugins
 
         super(CMSPlugin, cls).fix_tree(destructive)
         for placeholder in Placeholder.objects.all():
             for language, __ in settings.LANGUAGES:
-                order = CMSPlugin.objects.filter(placeholder_id=placeholder.pk, language=language,
-                                                 parent_id__isnull=True
-                                                 ).values_list('pk', flat=True)
+                order = CMSPlugin.objects.filter(
+                        placeholder_id=placeholder.pk, language=language,
+                        parent_id__isnull=True
+                    ).order_by('position').values_list('pk', flat=True)
```

Comment:
indentation is a bit odd, I would match closing parenthesis with the variable declaration, nothing major though.